verification/sign.go

```diff
@@ -0,0 +1,189 @@
+// Licensed under the Apache-2.0 license
+
+package verification
+
+import (
+	"bytes"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/x509"
+	"errors"
+	"math/big"
+	"testing"
+)
+
+// Obtain and validate signature of asymmetric signing.
+func TestAsymmetricSigning(d TestDPEInstance, c DPEClient, t *testing.T) {
+	useSimulation := false
+	handle := getInitialContextHandle(d, c, t, useSimulation)
+	// Get digest size
+	profile, err := GetTransportProfile(d)
+	if err != nil {
+		t.Fatalf("Could not get profile: %v", err)
+	}
+
+	digestLen := profile.GetDigestSize()
+	if digestLen != 32 && digestLen != 48 {
+		t.Fatalf("[FATAL]: Unsupported digest size.")
+	}
+
+	// Validate asymmetric signature generated
+	testAsymmetricSigning(d, c, t, handle, digestLen)
+}
+
+// Check command fails in simulated context because this context does not allow signing.
+// This is because simulation context does not allow using context's private key.
+func TestSignSimulation(d TestDPEInstance, c DPEClient, t *testing.T) {
+	useSimulation := true
+	handle := getInitialContextHandle(d, c, t, useSimulation)
+	defer c.DestroyContext(handle, 0)
+
+	// Get digest size
+	profile, err := GetTransportProfile(d)
+	if err != nil {
+		t.Fatalf("Could not get profile: %v", err)
+	}
+
+	digestLen := profile.GetDigestSize()
+	if digestLen != 32 && digestLen != 48 {
+		t.Fatal("[FATAL]: Unsupported digest size.")
+	}
+
+	if _, err := c.Sign(handle, make([]byte, digestLen), SignFlags(0), make([]byte, digestLen)); err == nil {
```

Comment:
nit: Should we also check that Symmetric sign fails?

Reply:
I don't see this change. Do you need to push to the branch?

Reply:
I see the changes in line:48, do you expect some other change? Am I misunderstanding something?

Reply:
<img width="328" alt="image" src="https://github.com/chipsalliance/caliptra-dpe/assets/94533459/839a51b2-ffc3-430c-a77f-6f88d139e1ef">